lib/umwelt/client/clone.ex

```diff
@@ -0,0 +1,95 @@
+defmodule Umwelt.Client.Clone do
+  @moduledoc "Clone main process"
+
+  use GenServer
+  require Logger
+
+  alias Umwelt.Client
+
+  def start_link(_),
+    do:
+      GenServer.start_link(
+        __MODULE__,
+        %{phase_id: nil, port: nil},
+        name: __MODULE__
+      )
+
+  def init(state), do: {:ok, state}
+
+  def handle_cast({:pull, params}, _state) do
+    send(self(), :start_pulling)
+
+    {:noreply, params}
```

Comment:
There is no need to reinvent [`GenServer.handle_continue/2`](https://hexdocs.pm/elixir/GenServer.html#c:handle_continue/2), specifically taking into account you want `:start_pulling` to execute unconditionally here..

```suggestion
    {:noreply, params, {:continue, :start_pulling}}
```
And below 
```elixir
  def handle_continue(:start_pulling, state) do
    …
  end
```

Reply:
Fixed